src/main.ts

```diff
@@ -15,13 +16,27 @@ export async function run() {
       version = core.getInput('version');
     }
 
+    let arch = core.getInput('node-arch');
+
+    // if node-arch supplied but node-version is not
+    // if we don't throw a warning, the already installed x64 node will be used which is not probably what user meant.
+    if (arch && !version) {
+      core.warning(
+        '`node-arch` is provided but `node-version` is missing. This results in using an already installed x64 node which is not probably what you meant. To fix this, provide `node-arch` in combination with `node-version`'
+      );
```

Comment:
Could you please add an extra note to the README and action.yml files instead of this warning: 
```
The architecture is used in conjunction with the node-version input
```
I believe that this mention will be enough to notify customers about usage of the architecture input.

Reply:
I prefer to keep the warning. Good software should be self-contained and self-explaining. No documentation can replace the program warnings/error messages.

I can add a similar warning to the readme for the user's knowledge.

Reply:
The `setup-node` action is used for self-hosted runners as well and we can't predict what software is installed on them. Could you please update your warning message:
```
'`architecture` is provided but `node-version` is missing. In this configuration version of Node won't be changed.'
```

Reply:
Changed the message to consider self-hosted runners

https://github.com/actions/setup-node/pull/194/commits/d6fea3c137a52740107025d7995836b7b4889bbd